COMETwebapp/Components/Common/OpenTab.razor

```diff
@@ -0,0 +1,82 @@
+﻿<!------------------------------------------------------------------------------
+// Copyright (c) 2023-2024 Starion Group S.A.
+//
+//   Authors: Sam Gerené, Alex Vorobiev, Alexander van Delft, Jaime Bernar, Théate Antoine, João Rua
+//
+//   This file is part of CDP4-COMET WEB Community Edition
+//   The CDP4-COMET WEB Community Edition is the Starion Web Application implementation of ECSS-E-TM-10-25
+//   Annex A and Annex C.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+------------------------------------------------------------------------------->
+@namespace COMETwebapp.Components.Common
+@using COMETwebapp.Model
+@using COMET.Web.Common.Model
+@inherits OpenModel
+
+<DxFormLayout CaptionPosition="CaptionPosition.Vertical" CssClass="py-5 px-4">
+    <DxFormLayoutItem ColSpanLg="12">
+        <div style="display:flex; flex-direction: column; align-items: center;">
+            <FeatherSmile Size="42"
+                          Color="currentColor"
+                          StrokeWidth="1.8f"/>
+            <h5 class="font-weight-bold mt-1">You have no model selected</h5>
+            <span>Select a model to start working on it</span>
+        </div>
+    </DxFormLayoutItem>
+    <DxFormLayoutItem Caption="Model" ColSpanLg="12">
+        <DxComboBox Id="model-selection" Data="@(this.ViewModel.AvailableEngineeringModelSetups)"
+                    TextFieldName="@nameof(EngineeringModelSetup.Name)"
+                    @bind-Value="@(this.ViewModel.SelectedEngineeringModel)"
+                    NullText="Select an Engineering Model"/>
+    </DxFormLayoutItem>
+    <DxFormLayoutItem Caption="View" ColSpanLg="12">
+        <DxComboBox Id="view-selection" Data="@(Applications.ExistingApplications.OfType<TabbedApplication>())"
+                    TextFieldName="@nameof(Application.Name)"
+                    @bind-Value="@(this.ViewModel.SelectedApplication)"
+                    NullText="Select a View"/>
+    </DxFormLayoutItem>
+    <DxFormLayoutItem Caption="Domain" ColSpanLg="12">
+        <DxComboBox Id="domain-selection" Data="@(this.ViewModel.AvailablesDomainOfExpertises)"
+                    TextFieldName="@nameof(DomainOfExpertise.Name)"
+                    @bind-Value="@(this.ViewModel.SelectedDomainOfExpertise)"
+                    NullText="Select a Domain of Expertise"/>
+    </DxFormLayoutItem>
+
+    @if (!this.IsEngineeringModelView)
```

Comment:
In case of an EngineeringModelView, you should be able to select an open EngineeringModel. In case of a simple ApplicationView (like SiteDirectory info, to mangage users for example), you don't have to select anything. Also, I think that the component should propose you only things that are already open but have a button to open a model. The main button should be to open the tab, not open the model. Also, make sure that when selecting the DoE (which can be different than the one that is currently selected), you will have to switch the domain first

Reply:
Based on the figma, it seams that only 1 button is needed. Then, if we propose everything all the time, the VM should check all possible case (what if the iteration is already open, what if the DoE is not the same, what to do in case of an EngineeringModelView, ...). Iteraction with the session service should be done

Reply:
New commit includes support for more cases, as well as visual feedbacks 🤙